internal/repository/verify.go

```diff
@@ -23,59 +23,87 @@ import (
 // another is to create a new RSL entry for the current state.
 var ErrRefStateDoesNotMatchRSL = errors.New("Git reference's current state does not match latest RSL entry") //nolint:stylecheck
 
-func (r *Repository) VerifyRef(ctx context.Context, target string, latestOnly bool) error {
+func (r *Repository) VerifyRef(ctx context.Context, localRefName, remoteRefName string, latestOnly bool) error {
 	var (
 		expectedTip plumbing.Hash
 		err         error
 	)
 
 	slog.Debug("Identifying absolute reference path...")
-	target, err = gitinterface.AbsoluteReference(r.r, target)
+	localRefName, err = gitinterface.AbsoluteReference(r.r, localRefName)
 	if err != nil {
 		return err
 	}
 
-	slog.Debug(fmt.Sprintf("Verifying gittuf policies for '%s'", target))
+	if remoteRefName == "" {
+		remoteRefName = localRefName
+	} else {
+		// remote ref name is different
+		// **We must consider RSL entries that have remoteRefName rather than
+		// localRefName**
+		slog.Debug("Name of reference differs on remote, identifying absolute reference path...")
+		remoteRefName, err = gitinterface.AbsoluteReference(r.r, remoteRefName)
```

Comment:
One issue here: if the user only provides a short name, this will try and identify the full ref name based on what's in the local copy of the repository, and that'll fail as the remote ref is expressly called something else here. Two possible solutions:
a) we require the user to give us the full ref path
b) we assume it's a branch and/or try to see if we can prefix match what the localRefName has


Between the two, I lean towards the first solution, especially since the invocation of verify-ref is likely to be hidden away from the user with the transport or via command compatibility. At the transport level especially, we automatically have the full ref path.

The second can be really helpful in some cases, but can fall apart in confusing ways in others, I think. We can also add the second later after careful consideration but it'd be difficult to remove it...

Reply:
I believe that in any gittuf verification process, it's crucial to have steps involved, even if it means using a more complex method. It's important to know exactly what reference you are verifying, especially when it's based on pat clarity on the tern matching. Ambiguity in this process could lead to critical issues, such as verifying "refs/heads/f1" instead of "refs/f1." This level of clarity is essential regardless of whether it's being used in the transport or not. @patzielinski @adityasaky, do you think any other places in verification have any ambiguity? I know that VerifyTag reference uses AbsoulteRef, too, which could lead to the problems stated above.

Reply:
AbsoluteRef is inherently not a problem, IMO, when it's user facing, as it expands, for eg, `main` to `refs/heads/main` _only if it's actually in the repository_. The second option I proposed above would just "guess" that `some-branch` is in fact `refs/heads/some-branch` as it's not actually in the repository.

Reply:
the PR as it stands is in option A so I suggest we proceed. thanks for the discussion!